content/en/docs/concepts/architecture/garbage-collection.md

```diff
@@ -0,0 +1,147 @@
+
+### Foreground cascading deletion
+
+In foreground cascading deletion, the owner object first enters a *deletion in
+progress* state. In this state, the following happens to the owner object:
+
+  * The object is still visible through the API
+  * The object `deletionTimestamp` field is set to the time that the object was marked for deletion.
+  * The object `metadata.finalizers` field contains the value
+    `foregroundDeletion`.
+
+After the owner object enters the deletion in progress state, the garbage
+collector deletes the dependents. After deleting all the dependent objects, the
+garbage collector deletes the owner object. 
+
+During foreground cascading deletion, the only dependents that block owner
+deletion are those that have the `ownerReference.blockOwnerDeletion=true` field.
+
+### Background cascading deletion {#background-deletion}
+
+In background cascading deletion, the Kubernetes API server deletes the owner
+object immediately and the garbage collector cleans up the dependent objects in
+the background. By default, Kubernetes uses background cascading deletion unless
+you manually use foreground deletion or choose to orphan the dependent objects.
+
+### Orphaned dependents
+
+When Kubernetes deletes an owner object, the dependents left behind are called
+orphan objects. By default, Kubernetes deletes dependent objects. However, you
+can force the garbage collector to orphan dependent objects by setting the
+`--cascade=orphan` flag when deleting the owner object, or by setting
+`propagationPolicy` to `Orphan` when using the API. 
+
+## Garbage collection of unused containers and images {#containers-images}
+
+The {{<glossary_tooltip text="kubelet" term_id="kubelet">}} performs garbage collection on unused images every five minutes and
+on unused containers every minute. You should avoid using external garbage
+collection tools, as these can break the kubelet behavior and remove containers
+that should exist. 
+
+To configure options for unused container and image garbage collection, tune the
+kubelet using a [configuration file](/docs/tasks/administer-cluster/kubelet-config-file/)
+and change the parameters related to garbage collection using the
+[`KubeletConfiguration`](/docs/reference/config-api/kubelet-config.v1beta1/#kubelet-config-k8s-io-v1beta1-KubeletConfiguration)
+resource type.
+
+### Container image lifecycle
+
+Kubernetes manages the lifecycle of all images through imageManager, with the
+cooperation of cadvisor. The kubelet considers the following disk
+usage limits when making garbage collection decisions:
+
+  * `HighThresholdPercent`
+  * `LowThresholdPercent`
+
+Disk usage above the configured `HighThresholdPercent` value triggers garbage
+collection, which deletes images in order based on the last time they were used,
+starting with the oldest first. The kubelet garbage collector deletes images
+until disk usage reaches the `LowThresholdPercent` value.
+
+### Container image garbage collection
```

Comment:
Let's make the What's next section of https://k8s.io/docs/concepts/containers/images/ link here - that'd be really good.